src/struct.h

```diff
@@ -56,4 +63,26 @@ struct Tuple
 };
 
 std::ostream &operator<<(std::ostream &os, const TupleFields &t);
+
+class StructManager
+{
+public:
+  using StructMap = std::map<std::string, std::shared_ptr<Struct>>;
```

Comment:
Nit: I'm not a fan of single use typedefs. Unless `StructMap` is used in multiple places, I suggest remove it b/c it just adds an extra layer of things to look at when reading code